COM-601 Add functions to calculate testing indicator 1

tmvumbi2 opened the pull request:
Jira ticket: https://jembiprojects.jira.com/browse/COM-601

tmvumbi2 committed:
COM-601 Add functions to calculate testing indicator 1
Adds SQL functions to calculate testing indicator 1

lauravignoli commented on reportssql/testing_report_functions.sql:
```diff
@@ -1,5 +1,37 @@
+        patient pat
+    WHERE
+        patientGenderIs(pat.patient_id, p_gender) AND
+        getPatientHIVTestDateFromCounselingForm(pat.patient_id) > TIMESTAMPADD(MONTH, -3, CURDATE()) AND
```
@tmvumbi2 The spec says: **If the patients last HIV test is not <= 3 months ago**. So this condition should be **getPatientHIVTestDateFromCounselingForm(pat.patient_id) < TIMESTAMPADD(MONTH, -3, CURDATE())** right?

See test cases 3 and 4:

https://docs.google.com/spreadsheets/d/1drmjK2JUgSeAOraH_jDIuE-kqMsNwN-lhZ1Q6QkWGGw/edit#gid=2038467389

Also, the function name should probably be getPatientPreviousHIVTestDateFromCounselingForm and it should check that is part of group "HIV Testing and Counseling", otherwise it will pick up any other HIV test date.

lauravignoli commented on reportssql/testing_report_functions.sql:
```diff
@@ -1,5 +1,37 @@
+    WHERE
+        patientGenderIs(pat.patient_id, p_gender) AND
+        getPatientHIVTestDateFromCounselingForm(pat.patient_id) > TIMESTAMPADD(MONTH, -3, CURDATE()) AND
+        getPatientHIVTestDateFromCounselingForm(pat.patient_id) BETWEEN p_startDate AND p_endDate AND
```
This should be a different function. There's no date for the Final Test Result actually, maybe the test result obs date could be used to check that it was done in the reporting period?

lauravignoli commented on reportssql/testing_report_functions.sql:
```diff
@@ -1,5 +1,37 @@
+        getPatientHIVTestDateFromCounselingForm(pat.patient_id) > TIMESTAMPADD(MONTH, -3, CURDATE()) AND
+        getPatientHIVTestDateFromCounselingForm(pat.patient_id) BETWEEN p_startDate AND p_endDate AND
+        getPatientHIVTestResultFromCounselingForm(pat.patient_id) = p_hivResult AND
+        getTestintEntryPoint(pat.patient_id) = p_testingEntryPoint AND
```
@tmvumbi2 Please fix typo "getTestingEntryPoint"

tmvumbi2 committed:
COM-601 Fix testing report indicator 1 functions
Verifies that the patient previous HIV test is older than 3 months and  that the obs with Concept “HIV Test Date” group concept is "HIV Testing and Counseling".
Verifies that the value of the observation “Final Test Result” is either “Positive” or “Negative” and the observation date isbetween [reportStartDate] and [reportEndDate]

tmvumbi2 committed:
COM-601 Fix typo on a function name
Replaces getTestintEntryPoint by getTestingEntryPoint
